modules/starfish/2.0/starfish2.smk

```diff
@@ -0,0 +1,247 @@
+import itertools as it
+import string
+import copy
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["starfish"]`
+CFG = op.setup_module(
+    name = "starfish",
+    version = "2.0",
+    subdirectories = ["inputs", "starfish", "outputs"],
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _starfish_input_vcf,
+    _starfish_rename_output, 
+    _starfish_output_vcf,
+    _starfish_output_union,
+    _starfish_output_venn,
+    _starfish_all,
+
+##### GLOBAL VARIABLES #####
+
+# All variant callers from config
+callers = list(CFG["inputs"]["vcf"].keys())
+callers = [caller.lower() for caller in callers]
+
+# The name of the union VCF file
+union_vcf = copy.deepcopy(callers)
+union_vcf.append("union")
+union_vcf = "_".join(union_vcf)
+not_union_vcf = "!" + union_vcf
+
+assert len(callers) < 6, (
+    "Starfish can only handle a maximum of 6 input VCFs."
+)
+
+##### RULES #####
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _starfish_input_vcf:
+    input:
+        vcf = lambda w: config["lcr-modules"]["starfish"]["inputs"]["vcf"][w.caller] 
+    output:
+        vcf = CFG["dirs"]["inputs"] + "vcf/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}.{caller}.vcf.gz", 
+        tbi = CFG["dirs"]["inputs"] + "vcf/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}.{caller}.vcf.gz.tbi"
+    run:
+        op.relative_symlink(input.vcf, output.vcf), 
+        op.relative_symlink(input.vcf + ".tbi", output.tbi)
+
+
+# Run Starfish
+rule _starfish_run:
+    input:
+        vcfs = expand(
+            CFG["dirs"]["inputs"] + "vcf/{{seq_type}}--{{genome_build}}/{{tumour_id}}--{{normal_id}}--{{pair_status}}.{caller}.vcf.gz", 
+            caller = callers
+            ),
+        tbis = expand(
+            CFG["dirs"]["inputs"] + "vcf/{{seq_type}}--{{genome_build}}/{{tumour_id}}--{{normal_id}}--{{pair_status}}.{caller}.vcf.gz.tbi", 
+            caller = callers
+            ),
+        reference = reference_files("genomes/{genome_build}/sdf"),
+        starfish_script = CFG["inputs"]["starfish_script"]
+    output:
+        complete = touch(CFG["dirs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/starfish.complete"), 
+        venn = CFG["dirs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/venn.pdf"
+    log:
+        stdout = CFG["logs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/starfish_run.stdout.log",
+        stderr = CFG["logs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/starfish_run.stderr.log"
+    params:
+        opts = CFG["options"]["starfish_run"]
+    conda:
+        CFG["conda_envs"]["starfish"]
+    threads:
+        CFG["threads"]["starfish_run"]
+    resources:
+        **CFG["resources"]["starfish_run"]
```

Comment:
Is how this works documented anywhere?

Reply:
Pending the addition of `op.prepare_resources()` to the oncopipe package...